plans/smlbench2/main.go

```diff
@@ -0,0 +1,232 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"reflect"
+	"time"
+
+	utils "github.com/ipfs/testground/plans/smlbench2/utils"
+	iptb "github.com/ipfs/testground/sdk/iptb"
+	"github.com/ipfs/testground/sdk/runtime"
+	"github.com/ipfs/testground/sdk/sync"
+	"github.com/libp2p/go-libp2p-core/peer"
+	// shell "github.com/ipfs/go-ipfs-api"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/ipfs/go-cid"
+)
+
+const (
+	// sizeBytes = 10 * 1024 * 1024 // 10mb
+	sizeBytes = 100 * 1024 * 1024 // 100mb
+)
+
+var cidSubtree = &sync.Subtree{
+        GroupKey:    "cid",
+        PayloadType: reflect.TypeOf(&cid.Cid{}),
+        KeyFunc: func(val interface{}) string {
+                return "cid"
+        },
+}
+
+func main() {
+	runenv := runtime.CurrentRunEnv()
+	if runenv.TestCaseSeq < 0 {
+		panic("test case sequence number not set")
+	}
+
+	timeout := func() time.Duration {
+                if t, ok := runenv.IntParam("timeout_secs"); !ok {
+                        return 30 * time.Second
+                } else {
+                        return time.Duration(t) * time.Second
+                }
+        }()
+
+	watcher, writer := sync.MustWatcherWriter(runenv)
+        defer watcher.Close()
+
+	spec := iptb.NewTestEnsembleSpec()
+	spec.AddNodesDefaultConfig(iptb.NodeOpts{Initialize: true, Start: true}, "local")
+
+	ctx := context.Background()
+	ensemble := iptb.NewTestEnsemble(ctx, spec)
+	ensemble.Initialize()
+
+	localNode := ensemble.GetNode("local")
+
+	peerID, err := localNode.PeerID()
+	if err != nil {
+		runenv.Abort(err)
+	}
+	fmt.Println("Jim peerID", peerID)
+
+	swarmAddrs, err := localNode.SwarmAddrs()
+	if err != nil {
+		runenv.Abort(err)
+	}
+	fmt.Println("Jim swarmAddrs", swarmAddrs)
+
+	ID, err := peer.IDB58Decode(peerID)
+	if err != nil {
+		runenv.Abort(err)
+	}
+
+	addrs := make([]ma.Multiaddr, len(swarmAddrs))
+	for i, addr := range swarmAddrs {
+		multiAddr, err := ma.NewMultiaddr(addr)
+		if err != nil {
+			runenv.Abort(err)
+		}
+		addrs[i] = multiAddr
+	}
+	addrInfo := &peer.AddrInfo{ID, addrs}
+
+	seq, err := writer.Write(sync.PeerSubtree, addrInfo)
+	if err != nil {
+                runenv.Abort(err)
+        }
+        defer writer.Close()
+
+	client := localNode.Client()
```

Comment:
There is never a "remote" in this test as it always spawns a node for itself

Reply:
In the original test, there's an IPTB node called "adder" and another one called "getter". In this one, the node gets spawned before it knows what it's role is going to be, so I just called it "localNode" as it's local to the Docker container ... but it could be named anything. Maybe just calling it "node" is good? Naming things is hard.